### PR TITLE
Enable `arc4random(3)` on all supported BSDs and macOS/Darwin

### LINKED ISSUE #12607: Enable arc4random(3) on all supported BSDs
All BSD-based platforms supported by Crystal have `arc4random(3)`, add support for it where it was missing.

While here, also add LibC definition for `arc4random_uniform(3)`, which probably will be useful for https://github.com/crystal-lang/crystal/issues/10859 (wasm32 already has it).

### PR DESCRIPTION
**Pull request by dmgk**
All BSD-based platforms supported by Crystal have arc4random(3), add support for it where it was missing.

Fixes #12607

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

**Comment by straight-shoota** [comment]
Haha sorry, @Blacksmoke16 I completely swapped all your labels 😆 

Just for clarification: I'd see it as feature, not a bug because Random worked on all BSDs, this just enables using a better function (I guess missing that could be seen as a bug, but it's not hard broken). Also we've usually labeled issues about `Random` in `topic:stdlib:crypto`. And then I created a new label for generic BSD just to make it complete =)

**Comment by Blacksmoke16** [comment]
Haha all good, I was basing it off of the original issue. Updated that one to be feature and generic bsd as well.

File: src/crystal/system/random.cr [hunk 1]
```diff
@@ -14,7 +14,7 @@ end
   require "./wasi/random"
 {% elsif flag?(:linux) %}
   require "./unix/getrandom"
-{% elsif flag?(:openbsd) || flag?(:netbsd) %}
+{% elsif flag?(:bsd) %}
```
Comment (jgaskins) [inline src/crystal/system/random.cr:17]: I forgot to put this one in my PR to your branch

```suggestion
{% elsif flag?(:bsd) || flag?(:darwin) %}
```

**Review by straight-shoota — APPROVED** [review]

**Review by sdogruyol — APPROVED** [review]
Thank you @dmgk 🙏

**Pull request merged**